lib/manager/components/deployment/PeliasPanel.js

```diff
@@ -0,0 +1,236 @@
+// @flow
+
+import Icon from '@conveyal/woonerf/components/icon'
+import { connect } from 'react-redux'
+import React, { Component } from 'react'
+import {
+  Button,
+  Checkbox,
+  FormControl,
+  ListGroup,
+  ListGroupItem,
+  Panel
+} from 'react-bootstrap'
+
+import SelectFileModal from '../../../common/components/SelectFileModal'
+import { updateDeployment, uploadPeliasWebhookCsvFile } from '../../actions/deployments'
+import type { Deployment, Project } from '../../../types'
+
+type Props = {
+  deployment: Deployment,
+  project: Project,
+  updateDeployment: typeof updateDeployment,
+  uploadPeliasWebhookCsvFile: typeof uploadPeliasWebhookCsvFile
+}
+
+type State = {
+  fileToDeleteOnSuccesfulUpload: string | null,
+  peliasWebhookUrl: string
+}
+
+class PeliasPanel extends Component<Props, State> {
+  state = {
+    fileToDeleteOnSuccesfulUpload: null,
+    peliasWebhookUrl: ''
+  };
+
+  componentDidMount = () => {
+    const { deployment } = this.props
+    this.setState({
+    // Auto-populate blank field if project has webhook URL in it
+      peliasWebhookUrl: deployment.peliasWebhookUrl || ''
+    })
+  };
+
+  /**
+   *  Method fired when Pelias Update checkbox is checked
+   */
+  _onChangeUpdatePelias = () =>
+    this._updateDeployment({
+      peliasUpdate: !this.props.deployment.peliasUpdate
+    });
+
+  /**
+   * Updates state on input changes. Does not persist changes to deployment
+   */
+  _onChangeTextInput = (event: SyntheticInputEvent<HTMLInputElement>) => {
+    const { target } = event
+    const { value, id } = target
+    this.setState({ [id]: value })
+  };
+  /**
+   * Fires when input field is blurred, and persists state changes to the deployment by
+   * making a network request
+  */
+  _onBlurTextInput = (event: SyntheticInputEvent<HTMLInputElement>) => {
+    const { target } = event
+    const { value, id } = target
+    this._updateDeployment({ [id]: value })
```

Comment:
This makes a network request whenever I focus/blur the field, even if I haven't changed anything.

Reply:
Good catch thank you! Fixed in 4103b3a372e10a8d2413c6081b1988e41109905c